middleware.ts

```diff
@@ -0,0 +1,31 @@
+// write middleware file for nextjs
+import type { NextRequest } from "next/server"
```

Comment:
Small nit but I believe these 2 lines can be consolidated as

`import { NextResponse, type NextRequest } from "next/server"`